### DISCUSSION TITLE
An AMD user asks about ROCm

### DISCUSSION THREAD

**Discussion by veirant** [openai/whisper#105] [Ideas]
what do you think about adding support not only for CUDA but also for AMD ROCm so that even users without a suitable graphics card can transcribe at reasonable speed?
![2022-09-24-20-08-40](https://user-images.githubusercontent.com/46203688/192102877-0313b4b3-6973-4601-81d5-9309025ddbac.png)

**Comment by marcin-przywoski** [comment]
I'd be so glad if that would be implemented. My 5950x does not do as good of a job as potentially my 6900xt would

**Comment by minhngoc25a** [comment]
Whisper depends a lot on PyTorch to do the processing stuff. So when you see your GPUs used when running this software, that's not Whisper. That's Whisper's dependency, PyTorch, the machine learning component that yearns for computing powers.

**Reply by fcakyon** [reply]
Luckily, Pytorch supports not only CUDA but also ROCm (as given in the ss above)

**Reply by Blue-Hair-Miku** [reply]
I apologize for my ignorance, but does this mean that AMD GPU processing is possible here? How would I get it to work?

**Reply by minhngoc25a** [reply]
> I apologize for my ignorance, but does this mean that AMD GPU processing is possible here? How would I get it to work?

Use the ROCm build. It's Linux-only now, apparently. If you want a Windows build, you'll have to compile it yourself. Search around. Google is your best friend.

**Reply by minhngoc25a** [reply]
Apparently, the software that AMD released for GPU programming, ROCm, only supports Linux systems right now. Thus there is only PyTorch ROCm support for Linux. If you are able to use WSL version 2, though, you can use it after installing some drivers.

**Reply by veirant** [reply]
> > I apologize for my ignorance, but does this mean that AMD GPU processing is possible here? How would I get it to work?
> 
> Use the ROCm build. It's Linux-only now, apparently. If you want a Windows build, you'll have to compile it yourself. Search around. Google is your best friend.

As far as I see it, for now it's not enough just to download an appropriate version of PyTorch, since its appliance requires some very specific functions to be called in the actual code, despite them doing the same exact thing as the ones resorting to CUDA. So, based on the errors I get, when I try to use the app with PyTorch ROCm being installed, it checks if your VD is amongst the modern Nvidia's pieces and desides either CUDA or CPU will be used. Although it seems to be very easy to implement. Please correct me if I'm wrong. I personally have never touched Pytorch (but going to :) ).

**Reply by veirant** [reply]
> > I apologize for my ignorance, but does this mean that AMD GPU processing is possible here? How would I get it to work?
> 
> Use the ROCm build. It's Linux-only now, apparently. If you want a Windows build, you'll have to compile it yourself. Search around. Google is your best friend.

Have you succeed trying to make it work on an AMD card? May you attach some more details if so?

**Reply by GrahamboJangles** [reply]
> > > I apologize for my ignorance, but does this mean that AMD GPU processing is possible here? How would I get it to work?
> > 
> > 
> > Use the ROCm build. It's Linux-only now, apparently. If you want a Windows build, you'll have to compile it yourself. Search around. Google is your best friend.
> 
> Have you succeed trying to make it work on an AMD card? May you attach some more details if so?

Of course he hasn't. He'll just tell you how easy it is with some Googling, but wouldn't be able to do it himself

**Reply by minhngoc25a** [reply]
> > > > I apologize for my ignorance, but does this mean that AMD GPU processing is possible here? How would I get it to work?
> > > 
> > > 
> > > Use the ROCm build. It's Linux-only now, apparently. If you want a Windows build, you'll have to compile it yourself. Search around. Google is your best friend.
> > 
> > 
> > Have you succeed trying to make it work on an AMD card? May you attach some more details if so?
> 
> Of course he hasn't. He'll just tell you how easy it is with some Googling, but wouldn't be able to do it himself

Even with "some Googling," I wasn't able to pull it off. Unless you're some low-level genius that can write your own driver for Windows, this thing only supports Linux for now.

**Comment by NielsMayer** [comment]
See https://github.com/openai/whisper/discussions/55#discussioncomment-4504980 for some hints
